guides/contexts.md

```diff
@@ -8,1060 +8,1313 @@
+> Note: we are starting with the basics for modeling an ecommerce system. In practice, modeling such systems yields more complex relationships such as product variants, optional pricing, multiple currencies, etc. We'll keep things simple in this guide, but the foundations will give you a solid starting point to building such a complete system.
+
+Phoenix generated the web files as expected in `lib/hello_web/`. We can also see our context files were generated inside a `lib/hello/catalog.ex` file and our product schema in the directory of the same name. Note the difference between `lib/hello` and `lib/hello_web`. We have a `Catalog` module to serve as the public API for product catalog functionality, as well as an `Catalog.Product` struct, which is an Ecto schema for casting and validating product data. Phoenix also provided web and context tests for us, which we'll look at later. For now, let's follow the instructions and add the route according to the console instructions, in `lib/hello_web/router.ex`:
 
 ```elixir
   scope "/", HelloWeb do
     pipe_through :browser
 
     get "/", PageController, :index
-+   resources "/users", UserController
++   resources "/products", ProductController
   end
 ```
 
-With the new route in place, Phoenix reminds us to update our repo by running `mix ecto.migrate`. Let's do that now:
+With the new route in place, Phoenix reminds us to update our repo by running `mix ecto.migrate`, but first we need to make a few tweaks to the generated migration in `priv/repo/migrations/*_create_products.exs`:
 
-```console
-$ mix ecto.migrate
+```elixir
+  def change do
+    create table(:products) do
+      add :title, :string
+      add :description, :string
+-     add :price, :decimal
++     add :price, :decimal, precision: 15, scale: 6, null: false
+-     add :views, :integer
++     add :views, :integer, default: 0, null: false
 
-[info]  == Running Hello.Repo.Migrations.CreateUsers.change/0 forward
+      timestamps()
+    end
+```
+
+We modified our price column to a specific precision of 15, scale of 6, along with a not-null constraint. This ensures we store currency with proper precision for any mathematical operations we may perform. Next, we added a default value and not-null constraint to our views count. With our changes in place, we're ready to migrate up our database. Let's do that now:
```

Comment:
Might also be worth linking to a blogpost or some external resource on why it's important to store money as decimal this way, or if the user should consider storing them as cents as integer. As a precaution.